### ISSUE TITLE
ShopifyAPI::Order.new does not return any order.id on successfull creation

### ISSUE THREAD

**Issue by Fladdermuz** [Shopify/shopify-api-ruby#994]
# Issue summary

Write a short description of the issue here ↓
Connecting to store via private app type setup, call ShopifyAPI::Order.new and create an order. Order is created on store successfully. 
when calling 
`if order.save` 
under old API, I could get the order id of the newly created order by simply saying `order.id` also the order name `order.name` which would be something like #2201 etc. Example response says order id should be there, but not seeing it. 

 

## Expected behavior

What do you think should happen?
After saving an order, should return the id, name, etc. I can't specify order id myself, as I need store to assign new one dynamically. 

## Actual behavior

What actually happens?
order has no id or name in response. 

Tip: include an error message (in a `<details></details>` tag) if your issue is related to an error
Here is inspection of order: 

#<ShopifyAPI::Order:0x00007f241a3f4c70 @original_state={}, @custom_prefix=nil, @forced_nils={"tags"=>false, "buyer_accepts_marketing"=>false, "send_receipt"=>false, "line_items"=>false, "email"=>false, "note"=>false, "billing_address"=>false, "shipping_address"=>false, "financial_status"=>false, "discount_codes"=>false}, @aliased_properties={}, @session=#<ShopifyAPI::Auth::Session:0x00007f2419ecb348 @id="3xxxxx", @shop="xxxxx-dev.myshopify.com", @state=nil, @access_token="xxxxxxxxxxxxxxxxxx", @scope=#<ShopifyAPI::Auth::AuthScopes:0x00007f2419ecb078 @compressed_scopes=#<Set: {}>, @expanded_scopes=#<Set: {}>>, @associated_user_scope=nil, @expires=nil, @associated_user=nil, @is_online=false, @shopify_session_id=nil>, @client=#<ShopifyAPI::Clients::Rest::Admin:0x00007f241a3f41a8 @base_uri="https://xxxxxx-dev.myshopify.com", @base_uri_and_path="https://xxxxx-dev.myshopify.com/admin/api/2022-07", @headers={:"User-Agent"=>"Shopify API Library v11.0.1 | Ruby 2.7.2", :Accept=>"application/json", "X-Shopify-Access-Token"=>"xxxxxxxxxxxx", "Content-Type"=>"application/json"}>, @errors=#<ShopifyAPI::Rest::BaseErrors:0x00007f241a3efc48 @errors=[]>, @line_items=[{"variant_id"=>"39985607147608", "quantity"=>1}], @app_id=nil, @billing_address={"first_name"=>"MMMM222", "last_name"=>"CCC", "address1"=>"322 Vista Dr", "address2"=>"", "city"=>"Horizion City", "province"=>"TX", "country"=>"United States", "phone"=>"", "zip"=>"79928"}, @browser_ip=nil, @buyer_accepts_marketing=true, @cancel_reason=nil, @cancelled_at=nil, @cart_token=nil, @checkout_token=nil, @client_details=nil, @closed_at=nil, @created_at=nil, @currency=nil, @current_subtotal_price=nil, @current_subtotal_price_set=nil, @current_total_discounts=nil, @current_total_discounts_set=nil, @current_total_duties_set=nil, @current_total_price=nil, @current_total_price_set=nil, @current_total_tax=nil, @current_total_tax_set=nil, @customer=nil, @customer_locale=nil, @discount_applications=nil, @discount_codes=[{"code"=>"FREE", "amount"=>"100", "type"=>"percentage"}], @email="matthew@nutralinkusa.com", @estimated_taxes=nil, @financial_status="paid", @fulfillment_status=nil, @fulfillments=nil, @gateway=nil, @id=nil, @landing_site=nil, @location_id=nil, @name=nil, @note="Customer Claimed Lunch AMazon and was given  - Default Title", @note_attributes=nil, @number=nil, @order_number=nil, @order_status_url=nil, @original_total_duties_set=nil, @payment_details=nil, @payment_gateway_names=nil, @payment_terms=nil, @phone=nil, @presentment_currency=nil, @processed_at=nil, @processing_method=nil, @referring_site=nil, @refunds=nil, @shipping_address={"first_name"=>"MMMM222", "last_name"=>"Crowder", "address1"=>"320 Canyon Vista Dr", "address2"=>"", "city"=>"Horizion City", "province"=>"Texas", "country"=>"United States", "zip"=>"79928", "phone"=>""}, @shipping_lines=nil, @source_identifier=nil, @source_name=nil, @source_url=nil, @subtotal_price=nil, @subtotal_price_set=nil, @tags="FREE", @tax_lines=nil, @taxes_included=nil, @test=nil, @token=nil, @total_discounts=nil, @total_discounts_set=nil, @total_line_items_price=nil, @total_line_items_price_set=nil, @total_outstanding=nil, @total_price=nil, @total_price_set=nil, @total_shipping_price_set=nil, @total_tax=nil, @total_tax_set=nil, @total_tip_received=nil, @total_weight=nil, @updated_at=nil, @user_id=nil, @send_receipt=true>

## Steps to reproduce the problem

1. Create new order.
 

## Logs
If applicable, enable the logs as described in the README, and paste the relevant portion here.

## Specifications

- `shopify_api` version: 2022-07
- Shopify API version used (example: `2020-07`):

**Comment by yhontar** [comment]
@Fladdermuz have you found a reason or workaround? I have faced the same problem during the migration to the new API

**Comment by Fladdermuz** [comment]
@yhontar no. I have no idea now how to create a new order and get the id. No way to query Shopify and try to find the order as someone could have placed one seconds after it was created, so trying to get the last order id is useless. Previous versions did that and it makes perfect sense to simply return the order id when the object is saved.
